benchmarking/runs/priority_algorithm/default_parameters/runtimes.py

```diff
@@ -0,0 +1,150 @@
+                            ],
+                            ScenarioAttribute.GPA.value: [
+                                (Gpa.A, 0.5),
+                                (Gpa.B, 0.5),
+                            ],
+                        },
+                    )
+                )
+
+                for scenario_name, scenario in scenarios.items():
+                    artifact = SimulationSet(
+                        settings=SimulationSettings(
+                            num_teams=class_size // team_size,
+                            scenario=scenario,
+                            student_provider=student_provider,
+                            cache_key=f"priority_algorithm/default_parameters/runtimes/{scenario_name}/class_size_{class_size}",
+                        ),
+                        algorithm_set={
+                            AlgorithmType.PRIORITY: [
+                                PriorityAlgorithmConfig(
+                                    MAX_KEEP=30,
+                                    MAX_SPREAD=100,
+                                    MAX_ITERATE=250 * multiplier,
+                                    MAX_TIME=1000000,
+                                    name=f"config_{250 * multiplier}-100-30",
+                                )
+                                if "social" in scenario_name
+                                else PriorityAlgorithmConfig(
+                                    MAX_KEEP=15,
+                                    MAX_SPREAD=30,
+                                    MAX_ITERATE=30 * multiplier,
+                                    MAX_TIME=1000000,
+                                    name=f"config_{30 * multiplier}-30-15",
+                                ),
+                            ],
+                        },
+                    ).run(num_runs=num_trials)
+
+                    metrics = {
+                        "PrioritySatisfaction": PrioritySatisfaction(
+                            goals_to_priorities(scenario.goals),
+                            False,
+                        ),
+                        "AverageSocialSatisfaction": AverageSocialSatisfaction(
+                            metric_function=is_happy_team_1hp_friend,
+                        ),
+                    }
+                    insight_output = Insight.get_output_set(
+                        artifact, list(metrics.values())
+                    )
+                    print(
+                        f"{scenario_name.title()} with {class_size} students, {'normal' if multiplier == 1 else 'bigger'} run"
+                    )
+                    print(
+                        f"Runtime: {Insight.average_metric(insight_output, Insight.KEY_RUNTIMES)}"
+                    )
+                    for name in metrics.keys():
+                        print(f"{name}: {Insight.average_metric(insight_output, name)}")
+                    print()
+
+
+class OneDiversityScenario(Scenario):
```

Comment:
We should just decide now if we're cool with defining scenario in the runs like this

And/or what the pattern is. Is it like only reusable scenarios go in the scenario file and the runs files can contain all the specifically made scenarios?

Reply:
My logic for the scenario in the run file is that we end up creating so many, one for each run, that they are better of kept in the same file. The idea of the scenarios folder was to reuse common scenarios, but most of the time we were running very specific scenarios and I felt it would make a mess of the scenarios folder. This PR obviously isn't bad, but branch 321 has a ton more, so that's where I started that pattern.

An issue #381 has been created to address the docs change needed for this